tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Remoting/Client/HttpProcessAndSendIntegration.cs

```diff
@@ -0,0 +1,85 @@
+        MethodName = "ProcessAndSend",
+        ReturnTypeName = "System.Net.HttpWebRequest",
+        ParameterTypeNames = new[] { "System.Runtime.Remoting.Messaging.IMessage", "System.Runtime.Remoting.Channels.ITransportHeaders", "System.IO.Stream", },
+        MinimumVersion = RemotingIntegration.Major4,
+        MaximumVersion = RemotingIntegration.Major4,
+        IntegrationName = RemotingIntegration.IntegrationName)]
+
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    // ReSharper disable once InconsistentNaming
+    public class HttpProcessAndSendIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="msg">The incoming request message instance</param>
+        /// <param name="headers">The headers for the incoming request message</param>
+        /// <param name="inputStream">The stream for the incoming request message</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, IMessage msg, ITransportHeaders headers, Stream inputStream)
+        {
+            if (Tracer.Instance.InternalActiveScope is var scope)
+            {
+                return new CallTargetState(scope);
+            }
+
+            return CallTargetState.GetDefault();
+        }
+
+        /// <summary>
+        /// OnMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TReturn">Type of the response, in an async scenario will be T of Task of T</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="returnValue">HttpResponse message instance</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value</returns>
+        internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        {
+            if (state.Scope?.Span is Span span && span.Tags is HttpTags httpTags && returnValue is HttpWebRequest request)
+            {
+                var requestUri = request.RequestUri;
+                string resourceUrl = requestUri != null ? UriHelpers.CleanUri(requestUri, removeScheme: true, tryRemoveIds: true) : null;
+                string httpUrl = requestUri != null ? UriHelpers.CleanUri(requestUri, removeScheme: false, tryRemoveIds: false) : null;
+
+                span.ResourceName = $"{request.Method} {resourceUrl}";
```

Comment:
It would be nice if all these files had `#nullable enable`, but should we handle it here at least? I'm not sure what the impact of non-trimmed resources in the app is? Also added `ToUpperInvariant()`, but not sure if we should (we should probably cache it with the following usage I guess)

```suggestion
                span.ResourceName = string.IsNullOrEmpty(resourceUrl) ? request.Method.ToUpperInvariant() : $"{request.Method.ToUpperInvariant()} {resourceUrl}";
```

Reply:
Done in 65e9b25f0b62a0b40ae7950a82981aebc1e638fb